internal/generate/olm-catalog/csv_cmd_prompt.go

```diff
@@ -0,0 +1,92 @@
+// Copyright 2020 The Operator-SDK Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package olmcatalog
+
+import (
+	"strings"
+
+	olmapiv1alpha1 "github.com/operator-framework/api/pkg/operators/v1alpha1"
+	"github.com/operator-framework/operator-sdk/internal/util/projutil"
+)
+
+// InteractiveCSVCmd includes the list of CSV fields which would be asked
+// to the user while generating CSV.
+type InteractiveCSVCmd struct {
+	// DisplayName is the name of the crd.
+	DisplayName string
+	// Keyword is a list of keywords describing the operator.
+	Keywords []string
+	// Description of the operator. Can include the features, limitations or
+	// use-cases of the operator.
+	Description string
+	// Name of the publishing entity behind the operator.
+	ProviderName string
+	// URL related to the publishing entity behind the operator.
+	ProviderURL string
+	// Maintainers is the list of organizational entities maintaining the operator.
+	Maintainer []string
+}
+
+// generateInteractivePrompt generates the prompts for user to provide input to the CSV
+// fields.
+func (s *InteractiveCSVCmd) generateInteractivePrompt() {
+	s.DisplayName = projutil.GetRequiredInput("Displayname for the operator: ")
+	s.Keywords = projutil.GetStringArray("List of keywords for your operator in the format :- " +
+		"word1, word2, word3 ... : ")
+	s.Description = projutil.GetRequiredInput("Description for the operator: ")
+	s.ProviderName = projutil.GetRequiredInput("The name of the publishing entity behind the operator: ")
+	s.ProviderURL = projutil.GetOptionalInput("Any relevant URL for the provider name (if exists): ")
+	s.Maintainer = projutil.GetStringArray("List of maintainers and their emails in the format :- " +
+		"name1:email1, name2:email2 ... : ")
+}
+
+// fillUImetadata populates the CSV with the data obtained from the interactive
+// prompts which appear while generating CSV.
+func (s *InteractiveCSVCmd) fillUImetadata(csv *olmapiv1alpha1.ClusterServiceVersion) {
+	if s.DisplayName != "" {
+		csv.Spec.DisplayName = s.DisplayName
+	}
+
+	if len(s.Keywords) != 0 {
+		csv.Spec.Keywords = s.Keywords
+	}
+
+	if s.Description != "" {
+		csv.Spec.Description = s.Description
+	}
+
+	if len(s.Maintainer) != 0 {
+		maintainers := make([]olmapiv1alpha1.Maintainer, 0)
+		for _, entity := range s.Maintainer {
+			entityDetails := strings.Split(entity, ":")
+			if len(entityDetails) == 2 {
+				m := new(olmapiv1alpha1.Maintainer)
```

Comment:
Nit: don't use a pointer if you don't have to:
```suggestion
				m := olmapiv1alpha1.Maintainer{}
```
Ditto for `Provider`.